pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -3193,13 +3176,6 @@ func (wf *Workflow) GetTemplateByName(name string) *Template {
 	return nil
 }
 
-func (w *Workflow) GetTemplates() []Template {
-	return append(
-		w.GetExecSpec().Templates,
-		w.Status.GetStoredTemplates()...,
-	)
-}
-
```

Comment:
this function was only ever called by `HasArtifactGC()`. I also noticed that `w.GetExecSpec().Templates` and `w.Status.GetStoredTemplates()` were not mutually exclusive so appending the two together might be misleading if somebody tries to use this function in the future